packages/plugin-haskell/index.js

```diff
@@ -7,9 +7,17 @@ export default {
   resolve(path, [target]) {
     const [, user, repo] = path.split('/');
     const filePath = target.replace(/\./g, '/');
+    const basePath = path.split(/\/(src|lib|app|test)\//)[0];
+
     return [
+      `{BASE_URL}${basePath}/src/${filePath}.hs`,
+      `{BASE_URL}${basePath}/lib/${filePath}.hs`,
+      `{BASE_URL}${basePath}/app/${filePath}.hs`,
+      `{BASE_URL}${basePath}/test/${filePath}.hs`,
```

Comment:
This order is the actual loading order and I wonder if this needs to be moved right before `githubSearch`. OctoLinker does a HTTP Head request for every single url until the first returns `200`. Therefor the most common scenario needs to be checked first.

Reply:
Actually, scenario I added handles not only monorepo approach, but will also work fine with single project repos. I did some testing and it usually matches at first hit. Bonus point of this is that we browse current branch/commit and don't force master.

Reply:
Ahh I see, that sounds great. Can you provide me a link to a mono repo so I can have a look by myself?